service/storage_service.cc

```diff
@@ -677,6 +677,19 @@ future<> storage_service::topology_state_load() {
         }
     }
 
+    // As soon as a node joins token_metadata.topology we
+    // need to drop all its rpc connections with ignored_topology flag.
+    {
+        std::vector<future<>> futures;
+        tmptr->get_topology().for_each_node([&](const locator::node* n) {
+            const auto ep = n->endpoint();
+            if (ep != inet_address{} && !saved_tmpr->get_topology().has_endpoint(ep)) {
+                futures.push_back(remove_rpc_client_with_ignored_topology(ep));
+            }
+        });
+        co_await when_all_succeed(futures.begin(), futures.end()).discard_result();
+    }
```

Comment:
Shouldn't this be done only if there is a node in `join_group0` state?

Reply:
Why care about that? We want to drop connections if their peer wasn't part of `token_metadata.topology`, this is exactly what we're formally maintaining here. The node appears in topology -> drop connections for it.

Reply:
> Why care about that?

To optimize out the loop if not needed. May be it is premature optimization.

Reply:
> May be it is premature optimization.

I think so, it's just one pass over nodes, and it's simple.